src/main/java/org/elasticsearch/plugins/repository/ChainedPluginRepository.java

```diff
@@ -0,0 +1,133 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.plugins.repository;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import java.util.Collection;
+
+public class ChainedPluginRepository implements PluginRepository {
+
+    private final String name;
+
+    private final ImmutableMap<String, PluginRepository> repositories;
+
+    private ChainedPluginRepository(String name, ImmutableMap<String, PluginRepository> repositories) {
+        this.name = name;
+        this.repositories = repositories;
+    }
+
+    @Override
+    public String name() {
+        return name;
+    }
+
+    @Override
+    public Collection<PluginDescriptor> find(String name) {
+        ImmutableList.Builder<PluginDescriptor> plugins = ImmutableList.builder();
+
+        if (repositories != null) {
+            for (PluginRepository repository : repositories.values()) {
+                plugins.addAll(repository.find(name));
+            }
+        }
+        return plugins.build();
+    }
+
+    @Override
+    public PluginDescriptor find(String organisation, String name, String version) {
+        if (repositories != null) {
+            for (PluginRepository repository : repositories.values()) {
+                PluginDescriptor plugin = repository.find(organisation, name, version);
+                if (plugin != null) {
+                    return plugin;
+                }
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public Collection<PluginDescriptor> list() {
+        return list(new Filter() {
+            @Override
+            public boolean accept(PluginDescriptor plugin) {
+                // Accepts all plugins
+                return true;
+            }
+        });
+    }
+
+    @Override
+    public Collection<PluginDescriptor> list(Filter filter) {
+        ImmutableList.Builder<PluginDescriptor> plugins = ImmutableList.builder();
+
+        if (repositories != null) {
+            for (PluginRepository repository : repositories.values()) {
+                plugins.addAll(repository.list(filter));
+            }
+        }
+        return plugins.build();
+    }
+
+    @Override
+    public void install(PluginDescriptor plugin) {
+        if (repositories != null) {
+            for (PluginRepository repository : repositories.values()) {
+                repository.install(plugin);
+            }
+        }
+    }
+
+    @Override
+    public void remove(PluginDescriptor plugin) {
+        if (repositories != null) {
+            for (PluginRepository repository : repositories.values()) {
+                repository.remove(plugin);
```

Comment:
Do we need to call it for each repo? I guess that once we have removed a plugin, we don't need to remove it again and again?